service/interface.go

```diff
@@ -0,0 +1,20 @@
+package service
+
+import (
+	"github.com/ahab94/ApnaSchool/db"
+	"github.com/ahab94/ApnaSchool/db/mongo"
+)
+
+// Service initializes our database instance
+type Service struct {
+	db db.DataStore
+}
+
+// NewService creates a connection to our database
+func NewService() (*Service, error) {
+	store, err := mongo.NewClient(db.Option{})
```

Comment:
Lets not create mongo here... We create this store in `runtime.go` and pass it in ` func NewService( ds db.Datastore) (*Service, error)`

Reply:
This way we won't have to handle error in this function. It will sort of minimize this operation.